src/threads/thread.cpp

```diff
@@ -433,6 +433,12 @@ int hpxc_thread_create(hpxc_thread_t* thread, hpxc_thread_attr_t const* attr,
     return 0;
 }
 
+int hpxc_yield()
+{
+    hpx::this_thread::suspend();
```

Comment:
I think you should use `hpx::this_thread::yield();` instead.

Reply:
Does hpx::this_thread::yield() yield the os/kernel thread?

Reply:
> @hkaiser Doesn't hpx::this_thread::yield() yield the os/kernel thread? Or do they both give control back to the scheduler?

Both give control to the HPX scheduler, `suspend` just -- well -- suspends (another thread has to resume the suspended thread), while `yield` suspends the current thread and adds it to the end of the scheduling queue so that it gets resumed at some point automatically.